src/groupeddataframe/grouping.jl

```diff
@@ -177,6 +213,318 @@ end
 Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
     GroupedDataFrame(gd.parent, gd.cols, gd.groups, gd.idx, gd.starts, gd.ends)
 
+"""
+    groupindices(gd::GroupedDataFrame)
+
+Return a vector of group indices for each row of `parent(gd)`.
+
+Rows appearing in group `gd[i]` are attributed index `i`. Rows not present in
+any group are attributed `missing` (this can happen if `skipmissing=true` was
+passed when creating `gd`, or if `gd` is a subset from a larger [`GroupedDataFrame`](@ref)).
+"""
+groupindices(gd::GroupedDataFrame) = replace(gd.groups, 0=>missing)
+
+"""
+    groupvars(gd::GroupedDataFrame)
+
+Return a vector of column names in `parent(gd)` used for grouping.
+"""
+groupvars(gd::GroupedDataFrame) = _names(gd)[gd.cols]
+
+# Get grouping variable index by its name
+function _groupvar_idx(gd::GroupedDataFrame, name::Symbol, strict::Bool)
+    i = findfirst(==(name), groupvars(gd))
+    i === nothing && strict && throw(ArgumentError("$name is not a grouping column"))
+    return i
+end
+
+"""
+    groupvalues([T::Type], gd::GroupedDataFrame)
+    groupvalues([T::Type], gd::GroupedDataFrame, i)
+    groupvalues(gd::GroupedDataFrame, i, col)
+
+Get the values of the grouping columns for `gd`, optionally specifying the group
+index `i` and grouping column `col`. `col` may be an integer index (of the
+grouping columns passed to [`groupby`](@ref)) or the column name itself. The
+optional first argument to the first two forms specifies the return type, which
+may be `NamedTuple` (default) or `Tuple`.
+
+### Returns
```

Comment:
Same here - do we need `### Returns` section (please consider if we can handle it in the main body of the docstring), as this would shorten it which is desirable (if we do not loose precision).

Also - in the whole PR - can you please change `###` to `#` in docstrings for headings as this is a standard we use now in DataFrames.jl (and it is consistent with Base)?